core/src/test/java/io/questdb/test/std/datetime/microtime/TimestampFormatCompilerTest.java

```diff
@@ -184,54 +219,54 @@ public void testFormatHour23OneDigit() throws Exception {
+        assertFormat("pm, 12", "a, k", "2017-03-31T12:00:00.000Z");
+        assertFormat("am, 3", "a, k", "2017-03-31T03:00:00.000Z");
+        assertFormat("am, 11", "a, k", "2017-03-31T11:59:59.999Z");
 
-        assertFormat("15", "k", "2017-03-31T14:00:00.000Z");
-        assertFormat("13", "k", "2017-03-31T12:00:00.000Z");
-        assertFormat("4", "k", "2017-03-31T03:00:00.000Z");
-        assertFormat("12", "k", "2017-03-31T11:59:59.999Z");
+        assertFormat("14", "k", "2017-03-31T14:00:00.000Z");
+        assertFormat("12", "k", "2017-03-31T12:00:00.000Z");
+        assertFormat("3", "k", "2017-03-31T03:00:00.000Z");
+        assertFormat("11", "k", "2017-03-31T11:59:59.999Z");
     }
 
     @Test
     public void testFormatHourTwelve() throws Exception {
-        assertFormat("pm, 03", "a, hh", "2017-03-31T14:00:00.000Z");
-        assertFormat("pm, 01", "a, hh", "2017-03-31T12:00:00.000Z");
-        assertFormat("am, 04", "a, hh", "2017-03-31T03:00:00.000Z");
-        assertFormat("am, 12", "a, hh", "2017-03-31T11:59:59.999Z");
+        assertFormat("pm, 02", "a, hh", "2017-03-31T14:00:00.000Z");
+        assertFormat("pm, 12", "a, hh", "2017-03-31T12:00:00.000Z");
+        assertFormat("am, 03", "a, hh", "2017-03-31T03:00:00.000Z");
+        assertFormat("am, 11", "a, hh", "2017-03-31T11:59:59.999Z");
 
-        assertFormat("03", "hh", "2017-03-31T14:00:00.000Z");
-        assertFormat("01", "hh", "2017-03-31T12:00:00.000Z");
-        assertFormat("04", "hh", "2017-03-31T03:00:00.000Z");
-        assertFormat("12", "hh", "2017-03-31T11:59:59.999Z");
+        assertFormat("02", "hh", "2017-03-31T14:00:00.000Z");
+        assertFormat("12", "hh", "2017-03-31T12:00:00.000Z");
+        assertFormat("03", "hh", "2017-03-31T03:00:00.000Z");
+        assertFormat("11", "hh", "2017-03-31T11:59:59.999Z");
     }
 
     @Test
     public void testFormatHourTwelveOneDigit() throws Exception {
-        assertFormat("pm, 3", "a, h", "2017-03-31T14:00:00.000Z");
-        assertFormat("pm, 1", "a, h", "2017-03-31T12:00:00.000Z");
-        assertFormat("am, 4", "a, h", "2017-03-31T03:00:00.000Z");
-        assertFormat("am, 12", "a, h", "2017-03-31T11:59:59.999Z");
+        assertFormat("pm, 2", "a, h", "2017-03-31T14:00:00.000Z");
+        assertFormat("pm, 12", "a, h", "2017-03-31T12:00:00.000Z");
+        assertFormat("am, 3", "a, h", "2017-03-31T03:00:00.000Z");
+        assertFormat("am, 11", "a, h", "2017-03-31T11:59:59.999Z");
 
-        assertFormat("3", "h", "2017-03-31T14:00:00.000Z");
-        assertFormat("1", "h", "2017-03-31T12:00:00.000Z");
-        assertFormat("4", "h", "2017-03-31T03:00:00.000Z");
-        assertFormat("12", "h", "2017-03-31T11:59:59.999Z");
+        assertFormat("2", "h", "2017-03-31T14:00:00.000Z");
+        assertFormat("12", "h", "2017-03-31T12:00:00.000Z");
+        assertFormat("3", "h", "2017-03-31T03:00:00.000Z");
+        assertFormat("11", "h", "2017-03-31T11:59:59.999Z");
```

Comment:
Would be good to see how `00:00:00.000Z` is converted to am/pm

Reply:
@ideoma done